pkg/controller/common/service_control.go

```diff
@@ -71,6 +71,11 @@ func needsRecreate(expected, reconciled *corev1.Service) bool {
 		return true
 	}
 
+	// LoadBalancerClass is immutable once set if target type is load balancer
+	if expected.Spec.Type == corev1.ServiceTypeLoadBalancer && expected.Spec.LoadBalancerClass != reconciled.Spec.LoadBalancerClass {
```

Comment:
I think these are pointers, not values.

```suggestion
	if expected.Spec.Type == corev1.ServiceTypeLoadBalancer && !reflect.DeepEqual(expected.Spec.LoadBalancerClass, reconciled.Spec.LoadBalancerClass) {
```

Reply:
🤦 good catch